src/databricks/labs/remorph/upgrades/v0.4.0_add_main_table_operation_name_column.py

```diff
@@ -0,0 +1,30 @@
+# pylint: disable=invalid-name,unused-argument
+import logging
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.remorph.config import ReconcileConfig
+from databricks.labs.remorph.contexts.application import ApplicationContext
+from databricks.labs.remorph.deployment.recon import RECON_JOB_NAME
+from databricks.labs.remorph.helpers import db_sql
+
+logger = logging.getLogger(__name__)
+
+
+def _upgrade_reconcile_metadata_main_table(installation: Installation, ws: WorkspaceClient):
+    reconcile_config = installation.load(ReconcileConfig)
+    table_identifier = f"{reconcile_config.metadata_config.catalog}.{reconcile_config.metadata_config.schema}.main"
+    sql = f"ALTER TABLE {table_identifier} ADD COLUMN operation_name  STRING AFTER report_type"
```

Comment:
how do we tackle manual interventions or table corruption?

Reply:
Handled